src/Explorer/Graph/NewVertexComponent/NewVertexComponent.tsx

```diff
@@ -0,0 +1,199 @@
+import { Dropdown, IDropdownOption, Stack, TextField } from "office-ui-fabric-react";
+import React, { FunctionComponent, useEffect, useState } from "react";
+import AddIcon from "../../../../images/Add-property.svg";
+import DeleteIcon from "../../../../images/delete.svg";
+import { NormalizedEventKey } from "../../../Common/Constants";
+import { GremlinPropertyValueType, InputPropertyValueTypeString, NewVertexData } from "../../../Contracts/ViewModels";
+import { EditorNodePropertiesComponent } from "../GraphExplorerComponent/EditorNodePropertiesComponent";
+
+export interface INewVertexComponentProps {
+  newVertexDataProp: NewVertexData;
+  partitionKeyPropertyProp: string;
+  onChangeProp: (labelData: NewVertexData) => void;
+}
+
+export const NewVertexComponent: FunctionComponent<INewVertexComponentProps> = ({
+  newVertexDataProp,
+  partitionKeyPropertyProp,
+  onChangeProp,
+}: INewVertexComponentProps): JSX.Element => {
+  const DEFAULT_PROPERTY_TYPE = "string";
+  const [newVertexData, setNewVertexData] = useState<NewVertexData>(newVertexDataProp || { label: "", properties: [] });
+  const propertyTypes: string[] = EditorNodePropertiesComponent.VERTEX_PROPERTY_TYPES;
+
+  useEffect(() => {
+    addNewVertexProperty(partitionKeyPropertyProp);
+    //eslint-disable-next-line
+  }, []);
+  useEffect(() => {
+    onChangeProp(newVertexData);
+  });
```

Comment:
Why are these in useEffect? I can't understand the purpose of that

Reply:
The purpose was to call the function once while mounting the component and second one for every render. However, now I've removed both the useEffect and initialized the values in useState itself. For second one, I've called the function as and when needed without impacting functionality. It is working as it is.